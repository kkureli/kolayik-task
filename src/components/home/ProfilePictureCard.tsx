import React, {FC} from 'react';
import {Image, StyleSheet, View} from 'react-native';
import {User} from '../../models';
import Typography from '../common/Typography';

interface PPCardProps {
  user: User;
}

const ProfilePictureCard: FC<PPCardProps> = ({user: {image, title, name}}) => {
  return (
    <View style={styles.container}>
      <Image style={styles.avatar} source={image} />
      <Typography h3>{name}</Typography>
      <Typography subtitle gray>
        {title}
      </Typography>
    </View>
  );
};

export default ProfilePictureCard;

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    marginTop: 10
  },
  avatar: {
    width: 150,
    height: 150,
    resizeMode: 'contain'
  }
});
