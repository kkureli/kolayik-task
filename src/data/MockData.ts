import {ContactMethods, SubMenuModel, User} from '../models';
const avatar = require('../assets/images/Avatar.png');

const mockUser: User = {
  image: avatar,
  name: 'Adnan Burhan',
  title: 'Product Designer',
  email: 'adnan@burhan.com',
  phoneNumber: 5360700195,
  department: 'Product',
  location: 'Istanbul',
  employmentStartDate: '25 Jan 2019',
  manager: 'Yunus Türkoğlu'
};

const contactOptions: ContactMethods[] = [
  {
    title: 'Email',
    icon: require('../assets/icons/contactOptions/emailIcon.png'),
    id: 'email'
  },
  {
    title: 'Call',
    icon: require('../assets/icons/contactOptions/phoneCallIcon.png'),
    id: 'call'
  }
];

const subMenu: SubMenuModel[] = [
  {
    title: 'Requests',
    icon: require('../assets/icons/subMenuIcons/requestsIcon.png'),
    id: '1',
    onPress: () => console.log('navigation.navigate(RequestsScreen)')
  },
  {
    title: 'Information',
    icon: require('../assets/icons/subMenuIcons/infoIcon.png'),
    id: '2',
    onPress: () => console.log('navigation.navigate(InformationScreen)')
  },
  {
    title: 'Career',
    icon: require('../assets/icons/subMenuIcons/careerIcon.png'),
    id: '3',
    onPress: () => console.log('navigation.navigate(InformationScreen)')
  },
  {
    title: 'Leaves',
    icon: require('../assets/icons/subMenuIcons/leavesIcon.png'),
    id: '4',
    onPress: () => console.log('navigation.navigate(LeavesScreen)')
  },
  {
    title: 'Assets',
    icon: require('../assets/icons/subMenuIcons/assetsIcon.png'),
    id: '5',
    onPress: () => console.log('navigation.navigate(AssetsScreen)')
  },
  {
    title: 'Documents',
    icon: require('../assets/icons/subMenuIcons/documentsIcon.png'),
    id: '6',
    onPress: () => console.log('navigation.navigate(DocumentsScreen)')
  }
];

export {mockUser, subMenu, contactOptions};
