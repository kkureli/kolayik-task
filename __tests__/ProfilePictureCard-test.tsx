import React from 'react';
import renderer from 'react-test-renderer';

import ProfilePictureCard from '../src/components/home/ProfilePictureCard';
import {mockUser} from '../src/data/MockData';

test('ProfilePictureCard renders correctly', () => {
  const tree = renderer.create(<ProfilePictureCard user={mockUser} />).toJSON();
  expect(tree).toMatchSnapshot();
});
