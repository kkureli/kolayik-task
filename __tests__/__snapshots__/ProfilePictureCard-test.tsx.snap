// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePictureCard renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": 10,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/Avatar.png",
      }
    }
    style={
      Object {
        "height": 150,
        "resizeMode": "contain",
        "width": 150,
      }
    }
  />
  <Text
    margin={null}
    padding={null}
    style={
      Object {
        "color": "#000000",
        "fontFamily": "Roboto",
        "fontSize": 20,
        "fontWeight": "normal",
        "letterSpacing": 0.15,
      }
    }
  >
    Adnan Burhan
  </Text>
  <Text
    margin={null}
    padding={null}
    style={
      Object {
        "color": "#535453",
        "fontFamily": "Roboto",
        "fontSize": 14,
        "fontWeight": "normal",
      }
    }
  >
    Product Designer
  </Text>
</View>
`;
